artcommon/artcommonlib/util.py

```diff
@@ -82,3 +86,55 @@ def merge_objects(a, b):
             # move new entry to the beginning
             c.move_to_end(k, last=False)
     return c
+
+
+def is_future_release_date(date_str):
+    """
+    If the input date is in future then return True elase False
+    """
+    try:
+        target_date = datetime.strptime(date_str, "%Y-%m-%d")
+    except ValueError:
+        return False
+    current_date = datetime.now()
+    if target_date > current_date:
+        return True
+    else:
+        return False
+
+
+def get_assembly_release_date(assembly, group):
+    """
+    Get assembly release release date from release schedule API
+    """
+    release_schedules = requests.get(f'{RELEASE_SCHEDULES}/{group}.z/?fields=all_ga_tasks', headers={'Accept': 'application/json'})
+    for release in release_schedules.json()['all_ga_tasks']:
+        if assembly in release['name']:
+            assembly_release_data = datetime.strptime(release['date_start'], "%Y-%m-%d").strftime("%Y-%b-%d")
+            break
+    return assembly_release_data
+
+
+def get_inflight(assembly, group):
+    """
+    Get inflight release name from current assembly release
+    """
+    inflight_release = None
+    assembly_release_date = get_assembly_release_date(assembly, group)
+    match = re.fullmatch(r"openshift-(\d+).(\d+)", group)
+    if not match:
+        raise ValueError(f"Invalid group name: {group}")
+    previous_release = f"openshift-{int(match[1])}.{int(match[2])-1}"
+    release_schedules = requests.get(f'{RELEASE_SCHEDULES}/{previous_release}.z/?fields=all_ga_tasks', headers={'Accept': 'application/json'})
+    for release in release_schedules.json()['all_ga_tasks']:
+        is_future = is_future_release_date(release['date_start'])
```

Comment:
all_ga_tasks are ordered by date, so go through and get the most unreleased release, if no inflight release, like we preparing 4.11 and for 4.10 wich stop building it will be empty so inflight is None